homeassistant/components/adguard/__init__.py

```diff
@@ -200,6 +206,8 @@ def device_info(self) -> dict[str, Any]:
             },
             "name": "AdGuard Home",
             "manufacturer": "AdGuard Team",
-            "sw_version": self.hass.data[DOMAIN].get(DATA_ADGUARD_VERION),
+            "sw_version": self.hass.data[DOMAIN][self._entry.entry_id].get(
+                DATA_ADGUARD_VERSION
+            ),
```

Comment:
Fetching the version is an async method, this is why I didn't put in in init. I decided to provide the entry instead of the software version so that when we need more information from the config entry in the future we don't need to refactor that much.

Reply:
It would be completely fine by me if we put it in now and remove in in the near future with a better solution 👍